### PR TITLE
feat(jans-auth-server): improved Invalid Request message for Authorization Endpoint #2775

### LINKED ISSUE #2775: feat: Improve Invalid Request message for Authorization Endpoint
Currently the `invalid_request` message is static. 

 ```
  "authorize":[
        {
            "id":"invalid_request",
            "description":"The request is missing a required parameter, includes an unsupported parameter or parameter value, or is otherwise malformed.",
            "uri":null
        }

```

It would be much more helpful if we validate the Authorization request, and provide a more specific error message. Only if something totally unexected happens, should we return the default message. It would also be nice if we could detect the `Accept-Language:` header, and allow several default messages based on language.

### PR DESCRIPTION
**Pull request by yuriyz**
### Description

feat(jans-auth-server): improved Invalid Request message for Authorization Endpoint

#### Target issue
  
closes #2775

### Test and Document the changes
- [x] Static code analysis has been run locally and issues have been fixed
- [x] Relevant unit and integration tests have been added/updated

Please check the below before submitting your PR. The PR will not be merged if there are no commits that start with `docs:` to indicate documentation changes or if the below checklist is not selected.
- [x] **I confirm that there is no impact on the docs due to the code changes in this PR.**

## Summary by CodeRabbit

* **Bug Fixes**
  * Authorization parameter validation now returns precise error reasons; PKCE and FAPI-related errors provide clearer messages.

* **Tests**
  * Added comprehensive unit tests for authorization parameter validation (response types, nonce, prompt, FAPI rules).
  * Updated client-side tests to use a shared crypto context and JWKS during registration; adjusted encrypted/JWE authorization request tests.
  * Disabled one unstable redirect-URI regex test.

### CODE REVIEW

**Comment by mo-auto** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scan Engine |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/janssenproject/pr-checks/395bb6a4-8bb8-4622-8ba4-f94720b16ccd?refs=e45402b6a03d532a2dc50a5f068cbcf4f68ba4e5&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

File: jans-auth-server/client/src/test/java/io/jans/as/client/ws/rs/jarm/AuthorizationResponseModeQueryJwtResponseTypeTokenIdTokenEncryptedHttpTest.java [hunk 1]
```diff
@@ -359,17 +360,17 @@ public void authorizationRequestObjectAlgRSA15EncA128CBCPLUSHS256(
...
+                authorizationRequest,
+                SignatureAlgorithm.RS256,
+                cryptoContext.getKeyId(Algorithm.RS256),
+                KeyEncryptionAlgorithm.RSA1_5,
+                BlockEncryptionAlgorithm.A128CBC_PLUS_HS256,
+                serverKeyId,
+                jwks,
+                TestCryptoContext.getInstance().getCryptoProvider(),
+                Lists.newArrayList(),
+                Lists.newArrayList());
```
[the remark this answers is not in the document]
Comment (yuriyz) [reply jans-auth-server/client/src/test/java/io/jans/as/client/ws/rs/jarm/AuthorizationResponseModeQueryJwtResponseTypeTokenIdTokenEncryptedHttpTest.java:373]: This is ok, we want to assert response mode here.

File: jans-auth-server/client/src/test/java/io/jans/as/client/interop/SupportsReturningDifferentClaimsInIdTokenAndUserInfoEndpoint.java [hunk 2]
```diff
@@ -44,13 +44,16 @@ public void supportsReturningDifferentClaimsInIdTokenAndUserInfoEndpoint(
...
+        TestCryptoContext cryptoContext = TestCryptoContext.getInstance();
 
         // 1. Register client
         RegisterRequest registerRequest = new RegisterRequest(ApplicationType.WEB, "jans test app",
                 StringUtils.spaceSeparatedToList(redirectUris));
         registerRequest.setResponseTypes(responseTypes);
         registerRequest.setSectorIdentifierUri(sectorIdentifierUri);
         registerRequest.setScope(scopes);
+        registerRequest.setJwks(cryptoContext.getJwksAsString());
+        registerRequest.addCustomAttribute("jansInclClaimsInIdTkn", "true");
```
[the remark this answers is not in the document]
Comment (yuriyz) [reply jans-auth-server/client/src/test/java/io/jans/as/client/interop/SupportsReturningDifferentClaimsInIdTokenAndUserInfoEndpoint.java:56]: Test verifies difference between id and user claims as it's stated in method name. Ignore jansInclClaimsInIdTkn.

File: jans-auth-server/server/src/main/java/io/jans/as/server/model/authorize/AuthorizeParamsValidator.java [hunk 3]
```diff
@@ -34,35 +34,52 @@ private AuthorizeParamsValidator() {
...
+     * @param responseMode    The response mode.
+     * @return Returns null if validation passes, or a specific error reason if validation fails.
      */
-    public static boolean validateParams(List<ResponseType> responseTypes, List<Prompt> prompts, String nonce,
-                                         boolean fapiCompatibility, ResponseMode responseMode) {
+    public static String validateParamsWithReason(List<ResponseType> responseTypes, List<Prompt> prompts, String nonce,
+                                                  boolean fapiCompatibility, ResponseMode responseMode) {
+        if (responseTypes.isEmpty()) {
+            return "response_type is required";
+        }
```
[the remark this answers is not in the document]
Comment (yuriyz) [reply jans-auth-server/server/src/main/java/io/jans/as/server/model/authorize/AuthorizeParamsValidator.java:50]: Per the project guideline "don't add error handling for scenarios that can't happen — trust internal code and framework guarantees", I'd skip this fix. The current behavior (NPE on null) would correctly signal a programmer error if a future caller forgot the contract; a silent "response_type is required"
   would be misleading.

File: jans-auth-server/server/src/test/java/io/jans/as/server/model/authorize/AuthorizeParamsValidatorTest.java [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    @Test
+    public void validateParamsWithReason_whenResponseTypeIsEmpty_shouldReturnReason() {
+        List<ResponseType> responseTypes = Collections.emptyList();
+        List<Prompt> prompts = Collections.emptyList();
+
+        String reason = AuthorizeParamsValidator.validateParamsWithReason(responseTypes, prompts, "nonce123", false, ResponseMode.QUERY);
+
+        assertNotNull(reason);
+        assertEquals(reason, "response_type is required");
+    }
```
[the remark this answers is not in the document]
Comment (yuriyz) [reply jans-auth-server/server/src/test/java/io/jans/as/server/model/authorize/AuthorizeParamsValidatorTest.java:35]: Adding these tests is coupled to the prior finding: either we accept both (tests + null-safety), or neither. Being consistent with the previous decision: not adding these tests.

**Review by yuriyzz — APPROVED** [review]
